### PR TITLE
Add NcbiTaxonomySQlite class to describe NCBI Taxonomy SQLite databases

### PR DESCRIPTION
**Pull request by pvanheus**
The [rust-ncibtaxonomy](https://bioconda.github.io/recipes/rust-ncbitaxonomy/README.html?highlight=ncbitaxonomy#recipe-Recipe%20&#x27;rust-ncbitaxonomy&#x27;) (also published on crates.io as [ncbitaxonomy](https://crates.io/crates/ncbitaxonomy)) tools work with the NCBI Taxonomy database stored as a SQLite database. This provides a Galaxy type for them and corrects some typos in other similar classes code.

(P.S. there is inconsistency in the code base with regards to class names: SQlite or SQLite. I have gone with the name of the parent SQlite class.)

### CODE REVIEW

**Comment by bernt-matthias** [comment]
I remember that ete3 also stored ncbi taxonomy as sqlite. Is this the same?

**Comment by pvanheus** [comment]
> I remember that ete3 also stored ncbi taxonomy as sqlite. Is this the same?

No this is not the same. The ETE3 format uses a structure as per Option 1 [here](https://makandracards.com/makandra/45275-storing-trees-in-databases) whereas ncbitaxonomy uses Option 2, the "materialised tree" design which is optimised for lineage queries.

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1854,12 +1854,66 @@ def set_meta(self, dataset, overwrite=True, **kwd):
...
+            results = c.execute(taxons_query).fetchall()
+            if len(results) == 0:
+                raise Exception('could not count size of taxonomy table')
+            dataset.metadata.taxon_count = results[0][0]
+        except Exception as e:
+            log.warning("%s, set_meta Exception: %s", self, e)
+
+    def sniff(self, filename):
+        if super(NcbiTaxonomySQlite, self).sniff(filename):
+            table_names = frozenset({'__diesel_schema_migrations', 'taxonomy'})
```
Comment (bernt-matthias) [inline lib/galaxy/datatypes/binary.py:1884]: Is it exactly these two tables? I guess one wants the check as specific as possible. But I admit that `__diesel_schema_migrations` seems quite specific .. is it?

Comment (pvanheus) [reply lib/galaxy/datatypes/binary.py:1892]: Yes it is exactly these 2 tables. The funnily named `__diesel_schema_migrations` table is used by the Rust diesel interface to keep track of schema versions and migrations.

Comment (pvanheus) [reply lib/galaxy/datatypes/binary.py:1884]: Yes, the `__diesel_schema_migrations` table stores the version of the schema and info on migrations from the Rust diesel ORM. And then the `taxonomy` table stores the data.

**Review by mvdbeek — APPROVED** [review]

**Comment by mvdbeek** [comment]
@pvanheus could you add a small example databse to `lib/galaxy/datatypes/test` ?

**Comment by pvanheus** [comment]
> @pvanheus could you add a small example databse to `lib/galaxy/datatypes/test` as `ncbitaxonomy.sqlite`?

I have done so but called the file `test.ncbitaxonomy.sqlite` as the `ncbitaxonomy.sqlite` part is the file extension (borrowing the convention from existing sqlite datatypes).

**Comment by pvanheus** [comment]
The test failure https://app.circleci.com/pipelines/github/galaxyproject/galaxy/8452/workflows/5ea5e4b1-ce13-4abe-97ea-661ac44e0971/jobs/93782/steps seems to be javascript related so is presumably not related to this PR.

**Comment by bgruening** [comment]
@galaxybot please test

**Comment by bgruening** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
The taxonomy file has been sniffed as chira:
```
        else:
>           assert file_ext == PARENT_SNIFFER_MAP.get(expected_file_ext, expected_file_ext)
E           AssertionError: assert 'chira.sqlite' == 'ncbitaxonomy.sqlite'
E             - chira.sqlite
E             + ncbitaxonomy.sqlite
```

the mulled tests are failing because depot is down, the javascript unit tests fail presumably because the worker machines run out of memory.

**Review by mvdbeek — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @pvanheus!

**Comment by bgruening** [comment]
Thanks @mvdbeek!

**Pull request merged**